### PR TITLE
Credential stuffing/mass authentication hunting query

### PR DESCRIPTION
**Pull request by thmcelro**
Hunting query to detect potential spikes caused by mass-authentication techniques such as credential stuffing. Uses Kusto anomaly detection features to find spikes during 1 hour windows of the work week, further validation is done to detect if the number of failed authentication results (based on status codes related to username, password or MFA failure) spiked during the same window. 

As credential stuffing often uses credentials compromised in data dumps, and relies on password re-use, it's likely that at least some of these credentials will be invalidated due to password policies, or simply old and no longer in use, this should cause a failure spike alongside and overall attempt spike.

### CODE REVIEW

File: Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  | where series_decompose_anomalies_list_dcount_UserPrincipalName_ad_flag == 1
+  //Calculate the percentage change between the spike and the average users authenticating
+  | project TimeGenerated=todatetime(list_TimeGenerated), Hour=hour, WeekDay=list_day, AccountsAuthenticating=list_dcount_UserPrincipalName, AverageAccountsAuthenticatin=round(avg_dcount_UserPrincipalName, 0), PercentageChange = round  ((list_dcount_UserPrincipalName - avg_dcount_UserPrincipalName) / avg_dcount_UserPrincipalName * 100,   2)
+  | order by PercentageChange desc 
+  //As an additional feature we collect successful and unsuccessful logins during the 1h windows with anomalies
+  | join kind=inner(
+  SigninLogs 
+  | where TimeGenerated >= ago(timerange) 
+  | where ResultType == "0" 
+  | summarize Success=dcount(UserPrincipalName) by bin(TimeGenerated, 1h)
```
Comment (shainw) [inline Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml:46]: To help with context, I suggest we bring thru the associated UserPrincipal Names -
SuccessAccounts = make_set(UserPrincipalName),

File: Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  | join kind=inner(
+      SigninLogs 
+      | where TimeGenerated >= ago(timerange) 
+      //Failed sign-ins based on failed username/password combos or failed MFA
+      | where ResultType in ("50126", "50074", "50057", "51004") 
+      | summarize Failed=dcount(UserPrincipalName) by bin(TimeGenerated, 1h)
```
Comment (shainw) [inline Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml:52]: To help with context, I suggest we bring thru the associated UserPrincipal Names -
FailedAccounts = make_set(UserPrincipalName),

File: Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  ) on TimeGenerated
+  | project-away TimeGenerated1
+  | extend Total = Failed + Success
+  | project TimeGenerated, SuccessRate = round((toreal(Success) / toreal(Total)) *100) , round(FailureRate = (toreal(Failed) / toreal(Total)) *100)
```
Comment (shainw) [inline Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml:56]: Add SuccessAccounts and FailedAccounts to your project - , SuccessAccounts, FailedAccounts

File: Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+id: 528c1708-a67e-4e2f-b76d-d5e5e88a22aa
+name: Anomalous spike in user logins with increase in signin failure rate to detect Credential Stuffing and other mass authentcation attacks.
```
Comment (shainw) [inline Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml:2]: Can we shorten this name, this is what shows in the UX and generally we prefer 50 chars or less.  The description will help clarify further.

**Review by shainw — CHANGES_REQUESTED** [review]
In addition to specific comments, the filename and name inside the YAML should closely match.  Once you fix up the name length in the YAML, change the filename appropriately.

File: Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml (hunk 4 above)
Comment (thmcelro) [reply Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml:2]: Sorted, I didn't know this so will keep that in mind in future

File: Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml (hunk 3 above)
Comment (thmcelro) [reply Hunting Queries/SigninLogs/LoginSpikeWithIncreaseFailureRate.yaml:56]: Great point, should have brought these through.

**Review by shainw — APPROVED** [review]

**Pull request merged**